### PR TITLE
Add krbd:osd_request_timeout to the virtualization storage class

### PR DESCRIPTION
**Pull request by malayparida2000**
https://bugzilla.redhat.com/show_bug.cgi?id=2238957
https://issues.redhat.com/browse/RHSTOR-4521

### CODE REVIEW

File: controllers/storagecluster/storageclasses.go [hunk 1]
```diff
@@ -292,6 +292,7 @@ func newCephBlockPoolVirtualizationStorageClassConfiguration(initData *ocsv1.Sto
 	virtualizationStorageClassConfig.storageClass.ObjectMeta.Annotations["description"] = "Provides RWO and RWX Block volumes suitable for Virtual Machine disks"
 	virtualizationStorageClassConfig.storageClass.Parameters["mounter"] = "rbd"
 	virtualizationStorageClassConfig.storageClass.Parameters["mapOptions"] = "krbd:rxbounce"
+	virtualizationStorageClassConfig.storageClass.Parameters["mapOptions"] = "krbd:osd_request_timeout"
```
Comment (Madhu-1) [inline controllers/storagecluster/storageclasses.go:295]: This will override the above one, please add it as `"krbd:rxbounce;krbd:osd_request_timeout"`

Comment (malayparida2000) [reply controllers/storagecluster/storageclasses.go]: Thanks, Madhu, Updated. If it looks good please LGTM. Will try to get it merged today.

File: controllers/storagecluster/storageclasses.go [hunk 2]
```diff
@@ -291,7 +291,7 @@ func newCephBlockPoolVirtualizationStorageClassConfiguration(initData *ocsv1.Sto
 	virtualizationStorageClassConfig.storageClass.ObjectMeta.Name = generateNameForCephBlockPoolVirtualizationSC(initData)
 	virtualizationStorageClassConfig.storageClass.ObjectMeta.Annotations["description"] = "Provides RWO and RWX Block volumes suitable for Virtual Machine disks"
 	virtualizationStorageClassConfig.storageClass.Parameters["mounter"] = "rbd"
-	virtualizationStorageClassConfig.storageClass.Parameters["mapOptions"] = "krbd:rxbounce"
+	virtualizationStorageClassConfig.storageClass.Parameters["mapOptions"] = "krbd:rxbounce;krbd:osd_request_timeout"
```
Comment (Madhu-1) [inline controllers/storagecluster/storageclasses.go:294]: osd_request_timeout should not need any timeout value to be set?

Comment (malayparida2000) [reply controllers/storagecluster/storageclasses.go:294]: @nixpanic @idryomov ^^

Comment (idryomov) [reply controllers/storagecluster/storageclasses.go:294]: `osd_request_timeout` does need a value: an integer for timeout in seconds.  0 means no timeout.

Comment (malayparida2000) [reply controllers/storagecluster/storageclasses.go:294]: @nixpanic ^^

Comment (nixpanic) [reply controllers/storagecluster/storageclasses.go:294]: I guess a reasonable value for `osd_request_timeout` might be 120? This should not cause too many temporary interruptions, and a Pod (or VM) would still be able to detect problems in just a few minutes.

Comment (gqlo) [reply controllers/storagecluster/storageclasses.go:294]: I am wondering did we add `osd_request_timeout` parameter in the end?

Comment (malayparida2000) [reply controllers/storagecluster/storageclasses.go:294]: @gqlo No, This PR was closed as there was no progress in determining a suitable request timeout number.

**Pull request closed without merging**